src/main/java/org/codehaus/mojo/exec/ExecJavaMojo.java

```diff
@@ -279,6 +306,10 @@ public void run()
                         lookup.findStatic( bootClass, "main",
                                                  MethodType.methodType( void.class, String[].class ) );
                     
+                    if ( blockSystemExit )
+                    {
+                        System.setSecurityManager( new SystemExitManager( originalSecurityManager ) );
```

Comment:
Using this makes the mojo non-threadSafe and might introduce "funny" race-conditions. This should be at least documented.

Reply:
@TobiX, interesting. Please elaborate. The link does not help much. What kind of race conditions do you envision? And what could be worse than the whole Maven process just shutting down hard when dealing with processes using `System.exit()`?